internal/provider/vaultsecrets/data_source_vault_secrets_app.go

```diff
@@ -106,8 +106,20 @@ func (d *DataSourceVaultSecretsApp) Read(ctx context.Context, req datasource.Rea
 
 	openAppSecrets := map[string]string{}
 	for _, appSecret := range appSecrets {
-		secretName := appSecret.Name
-		openAppSecrets[secretName] = appSecret.Version.Value
+		switch {
```

Comment:
Would it be cleaner to switch on secret.Type instead?

Reply:
cleaner: yes

was concerned about the string type, are we going to switch to enum at some point?

Reply:
maybe 🤷